blaze-core/src/main/scala/org/http4s/blaze/util/ProcessWriter.scala

```diff
@@ -52,7 +53,7 @@ trait ProcessWriter {
   final private def go(p: Process[Task, ByteVector], stack: List[StackElem], cb: CBType): Unit = p match {
     case Emit(seq) if seq.isEmpty =>
       if (stack.isEmpty) writeEnd(ByteVector.empty).onComplete(completionListener(_, cb))
-      else go(Try(stack.head.apply(End).run), stack.tail, cb)
+      else Trampoline(go(Try(stack.head.apply(End).run), stack.tail, cb))
```

Comment:
I'm not familiar with the `Trampoline` data structure, but typically to trampoline a recursive computation, the entire computation must be trampolined (not just each step of the computation), which means, for example, that `go` would have to return `Trampoline`, and the computation would be executed at a new caller of `go` (maybe `go` -> `go0`, and then top-level `go` executes the trampoline).